reasoner/controller/NestedDisjunctionController.java

```diff
@@ -41,12 +42,17 @@ processorDriver, driver(), processorContext(), disjunction, bounds,
         );
     }
 
-    protected static class Processor extends DisjunctionController.Processor<Processor> {
+    protected static class Processor extends DisjunctionController.Processor<ConceptMap, Processor> {
 
         private Processor(Driver<Processor> driver,
                           Driver<NestedDisjunctionController> controller, Context context,
                           ResolvableDisjunction disjunction, ConceptMap bounds, Supplier<String> debugName) {
             super(driver, controller, context, disjunction, bounds, debugName);
         }
+
+        @Override
+        Stream<ConceptMap, ConceptMap> getOrCreateHubReactive(Stream<ConceptMap, ConceptMap> fanIn) {
+            return fanIn;
```

Comment:
do you need something like in the `RootDisjunctionController` that calls filter and distinct operation?

Reply:
Wasn't originally there.
Trade-off between work & memory. NestedDisjunctions are always in NegationControllers, who probably do the caching.

Reply:
hmm okay if it is still correct let's leave it as is

Reply:
also, it's better not to leave comments like this about "probably do caching" - it should be pretty quick to check and then have the full information to make a decision on

Reply:
Indeed. It does cache.